### PR TITLE
Migrate libaries to new .csproj and support Net Standard

### LINKED ISSUE #300: Move core libraries (Core, Server and Client) to .Net Standard
In order to consume these libraries on Xamarin. We need to add support for targeting .Net Standard 1.4 and up.

This task is a bit of a chore. but will greatly improve comparability with most platforms including Linux and macOS since it'll run under Xamarin or .Net Core.

As for the rest of the projects. They can remain as they are as they're targeting WPF which is Windows only.

### PR DESCRIPTION
**Pull request by NZSmartie**
Before consuming Buttplug libraries in Xamarin, support for .Net Standard is needed. See Issue #300

---
Closes #300

### CODE REVIEW

**Comment by NZSmartie** [comment]
Having to use `NLog` - `v5.0.0-beta11` as it supports .Net Standard 1.x hopefully they release a stable soon enough

File: Buttplug.Client/ButtplugWSClient.cs [hunk 1]
```diff
@@ -107,7 +106,7 @@ public ButtplugWSClient(string aClientName)
         {
             if (_ws != null && (_ws.State == WebSocketState.Connecting || _ws.State == WebSocketState.Open))
             {
-                throw new AccessViolationException("Already connected!");
```
Comment (NZSmartie) [inline Buttplug.Client/ButtplugWSClient.cs]: AccessViolationException is intended for unsafe or unmanaged code. InvalidOperationException is much more appropriate here. (See: https://docs.microsoft.com/en-us/dotnet/api/system.accessviolationexception?view=netframework-4.7.1&viewFallbackFrom=netstandard-1.4)

File: Buttplug.Core/ButtplugJsonMessageParser.cs [hunk 2]
```diff
@@ -21,38 +21,50 @@ public class ButtplugJsonMessageParser
         [NotNull]
         private readonly JsonSchema4 _schema;
 
-        public ButtplugJsonMessageParser([NotNull] IButtplugLogManager aLogManager)
```
Comment (NZSmartie) [inline Buttplug.Core/ButtplugJsonMessageParser.cs]: Making the log manager nullable is intended for the xamarin stuff , not really intended for this PR. Sorry >~<

File: Buttplug.Core/ButtplugJsonMessageParser.cs [hunk 3]
```diff
@@ -21,38 +21,50 @@ public class ButtplugJsonMessageParser
...
             _bpLogger = aLogManager.GetLogger(GetType());
-            _bpLogger.Info($"Setting up {GetType().Name}");
+            _bpLogger?.Info($"Setting up {GetType().Name}");
             IEnumerable<Type> allTypes;
 
             // Some classes in the library may not load on certain platforms due to missing symbols.
             // If this is the case, we should still find messages even though an exception was thrown.
             try
             {
+#if NETSTANDARD1_4
```
Comment (NZSmartie) [inline Buttplug.Core/ButtplugJsonMessageParser.cs]: These preprocessor directives aren't needed. Gonna remove them

File: Buttplug.Core/IButtplugLog.cs [hunk 4]
```diff
@@ -25,11 +25,5 @@ public interface IButtplugLog
 
         [CanBeNull]
         event EventHandler<LogExceptionEventArgs> OnLogException;
-
-        [NotNull]
-        Error LogErrorMsg(uint aId, ErrorClass aCode, string aMsg);
-
-        [NotNull]
-        Error LogWarnMsg(uint aId, ErrorClass aCode, string aMsg);
```
Comment (NZSmartie) [inline Buttplug.Core/IButtplugLog.cs]: Made these methods as Extension methods to reduce copying code and only needing to implement whjat's required

File: Buttplug.Server/ButtplugServer.cs [hunk 5]
```diff
@@ -67,8 +67,8 @@ public ButtplugServer([NotNull] string aServerName, uint aMaxPingTime, DeviceMan
             _pingTimedOut = false;
             if (aMaxPingTime != 0)
             {
-                _pingTimer = new Timer(_maxPingTime);
-                _pingTimer.Elapsed += PingTimeoutHandler;
+                // Create a new timer that wont fire any events jsut yet
```
Comment (qdot) [inline Buttplug.Server/ButtplugServer.cs]: nit: just

File: Buttplug.Core/ButtplugJsonMessageParser.cs [hunk 6]
```diff
@@ -38,16 +38,16 @@ public ButtplugJsonMessageParser([NotNull] IButtplugLogManager aLogManager)
                 allTypes = e.Types;
             }
 
-            var messageClasses = from t in allTypes
-                                 where t != null && t.IsClass && t.Namespace == "Buttplug.Core.Messages" && typeof(ButtplugMessage).IsAssignableFrom(t)
-                                 select t;
+            var messageClasses = allTypes
```
Comment (qdot) [inline Buttplug.Core/ButtplugJsonMessageParser.cs]: talked about this on IRC, but I guess you don't need LINQ expansion anymore?

**Pull request merged**